### PR TITLE
feat: build and push new prebuilt docker image

### PR DESCRIPTION
**Pull request by sammcj**
# Summary of changes

Fix for https://github.com/anothrNick/github-tag-action/issues/163

- Build, label, tag and publish Docker image to Githubs image repository.
- Simplify the Actions Dockerfile to pull the built image.
- (Fixes auth issues faced last PR! https://github.com/anothrNick/github-tag-action/pull/201)
- Add missing .gitignore (how did the project never have one?!).
- Use standard ubuntu-latest as this is what Github recommend now.

Do any of the followings changes break current behaviour or configuration?

- No

## How changes have been tested

- I've forked the repo and done the build and publish.
- Tested I'm able to pull the build image locally.

Note that tests on this PR won't pass as they're run from the main branch that doesn't yet have the ghcr image:

image

## List any unknowns

- I haven't tested the changes in _this_ repo.
	- I don't know what package permissions @anothrNick has setup on this repo, if it's default I believe public repos have public packages.

Once merged, the Action should be tested to ensure it's operating the same.

### CODE REVIEW

**Comment by sammcj** [comment]
@sbe-arg it might be good for you to also pull this and test it with whatever methods you've used in the past as well 👍

**Comment by sbe-arg** [comment]
Ill try to test this later today. I did similar work in another repo last week XD

File: .github/workflows/build-docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            type=semver,pattern={{major}}
+            type=sha
+
+      - name: Set up QEMU To support build amd64 and arm64 images
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to DockerHub
```
Comment (miguelaferreira) [inline .github/workflows/build-docker-image.yml:49]: It's not really logging in to DockerHub
```suggestion
      - name: Login to Docker repository
```

File: Dockerfile [hunk 2]
```diff
@@ -1,10 +1,8 @@
-FROM node:16-alpine
-LABEL "repository"="https://github.com/anothrNick/github-tag-action"
-LABEL "homepage"="https://github.com/anothrNick/github-tag-action"
-LABEL "maintainer"="Nick Sjostrom"
-
-RUN apk --no-cache add bash git curl jq && npm install -g semver
+# hadolint ignore=DL3007
+FROM ghcr.io/anothrnick/github-tag-action:latest
```
Comment (miguelaferreira) [inline Dockerfile:2]: How will this play out with versioning of the action? When a user just declares the action like this
```yaml
- name: Bump version and push tag
  uses: anothrNick/github-tag-action@1.55.0 # Don't use @master unless you're happy to test the latest version
  env:
    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
    WITH_V: true
```

Will GHA build version `1.55.0` of the action? Or will it be `ghcr.io/anothrnick/github-tag-action:latest`?

**Review by miguelaferreira — COMMENTED** [review]
I'm not a maintainer, just an interested user.

I'm very excited about this PR, not having to build the action on every workflow execution will speed things up, and make for more reliable and predictable workflows.

I do have a question about versioning. In this PR the Dockerfile pulls `latest`, and I'm wondering if that will play well with explicit versions.

File: Dockerfile (hunk 2 above)
Comment (sbe-arg) [reply Dockerfile:2]: If we don't retro build all past tags as images they will be built on demand by the local workflow

Comment (miguelaferreira) [reply Dockerfile:2]: And moving forward the tagged images will be available?

**Comment by sammcj** [comment]
Whoops fell behind on this PR, way too much going on at the mo. I'll try to get back to it maybe late this week.

The call out about versioning mentioned above is warranted. Might need to do some tweaks.

**Pull request closed without merging**